src/metabase/query_processor.clj

```diff
@@ -37,6 +38,7 @@
   [#'qp.middleware.enterprise/handle-audit-app-internal-queries-middleware
    #'process-userland-query/process-userland-query-middleware
    ;; userland queries only: catch Exceptions and return a special error response
+   #'update-used-cards/update-used-cards!
```

Comment:
Hmmm, maybe this should be execution middleware instead. `metabase.query-processor.execute/middleware`. We shouldn't be fetching more Cards after executing the query. We should try to avoid putting stuff in the general middleware here unless it really can't go in one of the more specific spots.

Also, the comment above this stopped making sense after you added this line, would have been better to add it above the comment instead. Comments are no good if they're not next to what they're talking about

Reply:
moved to execute middleware